### PR TITLE
Add Nougat OCR loader

### PR DESCRIPTION
**Pull request by mdarshad1000**
# Description

A PDF loader using the recently released Nougat-OCR loader which is the academic document PDF parser that understands LaTeX math and tables. I was looking for existing loaders. It can be used for ingesting equation-heavy papers.

## Type of Change

Please delete options that are not relevant.

- [x ] New Loader

# How Has This Been Tested?

Please describe the tests that you ran to verify your changes. Provide instructions so we can reproduce. Please also list any relevant details for your test configuration

- [x ] Added a new notebook illustrating the working

# Suggested Checklist:

- [x] I have added a library.json file if a new loader/tool was added
- [x] I have performed a self-review of my own code
- [x] I have commented on my code, particularly in hard-to-understand areas
- [ ] My changes generate no new warnings
- [x] I have added tests that prove my fix is effective or that my feature works
- [ ] New and existing unit tests pass locally with my changes

### CODE REVIEW

File: llama_hub/library.json [hunk 1]
```diff
@@ -955,5 +955,14 @@
       "repository",
       "collaborators"
     ]
+  },
+  "NougatOCRReader": {
+    "id": "github_repo_collaborators",
```
Comment (EmanuelCampos) [inline llama_hub/library.json:960]: id is duplicated

**Review by EmanuelCampos — COMMENTED** [review]
That's very cool. 

a README would be excellent, and did a comment with a small fix

**Review by jerryjliu — COMMENTED** [review]
exciting! some comments: 
- why is the implementation of PDFNougatOCR duplicated in the notebook? should just import from the llama-hub code (`from llama_hub.nougat_ocr.base import PDFNougatOCR`)
- is mathpaper.pdf downloadable? just so other users can download the pdf and run the notebook? (would prefer that vs. checking the pdf in) 
- can you clear the output of the document? it's too long to check-in

File: llama_hub/library.json (hunk 1 above)
Comment (mdarshad1000) [reply llama_hub/library.json:960]: Great, thanks.  I'll fix and push it asap.

**Comment by mdarshad1000** [comment]
* sure, I'll make the necessary changes to avoid duplication and ensure consistency.
* yes, I will add the mathpaper.pdf file so that other users can easily download it and run the notebook without any issues
* certainly, I will clear the output of the document to make it more manageable.

**Comment by jerryjliu** [comment]
@mdarshad1000 let me know when you get to this! would be excited to promote

**Comment by jerryjliu** [comment]
@mdarshad1000 quick bump on this

**Comment by mdarshad1000** [comment]
caught a flu, I'm on it

File: llama_hub/library.json [hunk 2]
```diff
@@ -983,5 +983,14 @@
...
```
Comment (jerryjliu) [inline llama_hub/library.json:988]: id should be "nougat_ocr"

**Review by jerryjliu — COMMENTED** [review]
almost there! there's some lint errors, please run `make format`

File: llama_hub/library.json (hunk 2 above)
Comment (EmanuelCampos) [reply llama_hub/library.json:988]: and NougatOCRReader should match the class name and readme PDFNougatOCR

**Comment by EmanuelCampos** [comment]
@logan-markewich opened an issue on nougat-ocr, for inference through code, avoiding call subprocress

https://github.com/facebookresearch/nougat/issues/142

**Comment by EmanuelCampos** [comment]
I had an issue running nougat on my Macbook M1 16gb, completely crashed the system and restarted, still investigating why. and not able to through the notebook yet, will do a few more tests soon

https://media.discordapp.net/attachments/1113531342563582062/1159941820839628850/image.png?ex=6532db00&is=65206600&hm=d6558acc7b4ec06666dc261daf9b95e5b3b8244f01b1840b09190b382d09d14c&

**Pull request merged**